kahuna/public/js/main.js

```diff
@@ -151,16 +151,29 @@ kahuna.run(['$rootScope', 'mediaApi',
 
 // Intercept 401s and emit an event
 kahuna.config(['$httpProvider', function($httpProvider) {
-    $httpProvider.interceptors.push('httpUnauthorisedInterceptor');
+    $httpProvider.interceptors.push('httpErrorInterceptor');
 }]);
 
-kahuna.factory('httpUnauthorisedInterceptor',
+kahuna.factory('httpErrorInterceptor',
                ['$q', '$rootScope', 'httpErrors',
                 function($q, $rootScope, httpErrors) {
     return {
         responseError: function(response) {
-            if (response.status === httpErrors.unauthorised.errorCode) {
-                $rootScope.$emit('events:error:unauthorised');
+            switch (response.status) {
+                case 0:
```

Comment:
I found it happened on POST failure - I changed the [shape of the payload](https://github.com/guardian/grid/blob/master/kahuna/public/js/services/label.js#L25) in the add label message in DEV to test this (`.post({data_: labels})` vs. `.post({data: labels})`). This caused a server error which returned 500 in the network tab, but angular always showed it as 0.